### ISSUE TITLE
Darker Theme

### ISSUE THREAD

**Issue by kaeh** [Xanthos-Eve/vintel#78]
If possible, a darker (night) theme would be welcome !

**Comment by Xanthos-Eve** [comment]
I agree - its really stark next to the dark eve client screens. I have done
a little investigation into this but so far its not as simple as it might
seem given some of the idiosyncrasies of QT.

Anyone with suggestions on how to implement this OR how it should look
please chime in. QT supports style sheets so, it could be made to look
really good - for instance, we could actually use a star field image or
pattern for the map background.

Feel free to experiment and share your findings.

**Comment by simonlinj** [comment]
+1

**Comment by kaeh** [comment]
o7 mateys,

Sorry for the long silence of me, i came back with three visual proposition i made today for this darker theme.

Dark Theme : http://imgur.com/mblGjnu
Lighter Dark Theme : http://imgur.com/jrNRtzl
Full Darky Theme : http://imgur.com/PwQKwxJ

Of course other colors would still be the same (red for dangerous, green for clear etc)

I'ld love to help developing this soft (i love programming), but i don't know shit in python...

**Comment by Xanthos-Eve** [comment]
Hey thanks! Fully dark looks best though it may be harder to do. There is
another aspect of this and that is the app widgets themselves. It will be
great to have a nice dark map screen but to complete the effect we will
also need the widow frame, buttons etc, to be themed as well.

**Comment by Xanthos-Eve** [comment]
Widgets, etc. use a style sheet so in theory this should be straight
forward. In practice there are a lot of edge cases to tweak. I put support
in vintel.py for the stylesheet. More work is needed.

**Comment by Xanthos-Eve** [comment]
Messed around with this a while back - there were problems with scroll bars as I recall.

https://github.com/ColinDuquesnoy/QDarkStyleSheet

**Comment by Xanthos-Eve** [comment]
The map is an svg file so using a style sheet there would be the way to of.

**Comment by eNTi** [comment]
Why don't you load an inverted version of the svg OR just invert it on the fly?

**Comment by Xanthos-Code** [comment]
How does one invert an svg?

**Comment by eNTi** [comment]
yeah i realized that this doesn't make much sense after i looked into your code. so you'd basically have to write a theme "engine" that reads some kind of config (theme) file?

**Comment by Xanthos-Code** [comment]
It can all be styled via css.

**Comment by thomaslener** [comment]
Maybe there's an easy solution for a Dark Theme. Vintel takes the map from an URL like http://evemaps.dotlan.net/svg/Catch.svg

Possible solution: dotlan serving an svg with dark background.

Since dotlan already has a "Dark Theme" feature implemented, it could be beneficial to talk to the creator of dotlan.

Xanthos, if it would help to get a dark themed svg from dotlan, i would contact the creator of dotlan to ask if this is possible. But i need your clearance first.

------------------------------------------------------------------------------------------------------------
Edit 22/02/2017:
Contacted the creator of Dotlan via Ingame Mail:

Sent: 2017.02.22 17:26
To: Wollari,  

Hi Wollari,

im am writing because VINTEL (https://github.com/Xanthos-Eve/vintel) uses DOTLAN svg (eg http://evemaps.dotlan.net/svg/Catch.svg) and VINTEL is in need of a Dark Theme in order to not hurt the users eyes at night.

Is it possible to serve a dark themed version for this svg file?

o7

**Comment by Sikksens** [comment]
There is already a dark svg available from dotlan, I'm working on a dark version now but my knowledge of Qt is very low, I've managed to import a dark svg from dotlan and changed around a few colours to accommodate the dark map.

http://i.imgur.com/VrxwfQi.png

I know @Xanthos-Eve was talking about making the whole app dark but I feel just the map and intel panel being dark is ok for now.

I do need help changing the colour of the text in the intel panel to white if anyone has an idea how, I will continue fiddling around with it more and hopefully have a semi-decent dark vintel soon.
